scripts/gen-keybinding-img

```diff
@@ -0,0 +1,424 @@
+        context.paint()
+        context.restore()
+
+    def render(self, filename):
+        surface = cairo.ImageSurface(cairo.FORMAT_ARGB32, 1280, 800)
+        context = cairo.Context(surface)
+        with context:
+            context.set_source_rgb(1, 1, 1)
+            context.paint()
+
+        self.add_logo(context)
+
+        context.move_to(210, 80)
+        context.set_font_size(28)
+        context.show_text('Keybindings for Qtile')
+
+        context.move_to(210, 100)
+        context.set_font_size(18)
+        if len([i for i in self.modifiers if i]):
+            context.show_text('Modifiers: ' + ', '.join(self.modifiers))
+        else:
+            context.show_text('No modifiers used.')
+
+        for i in self.key_pos.values():
+            if i.key in ['FN_KEYS']:
+                continue
+
+            self.draw_button(context, i.key, i.x, i.y, i.width, i.height)
+
+        # draw functional
+        fn = [i for i in keys.values() if i.key[:4] == 'XF86']
+        if len(fn):
+            fn_pos = self.key_pos['FN_KEYS']
+            x = fn_pos.x
+            for i in fn:
+                self.draw_button(context, i.key, x, fn_pos.y, fn_pos.width, fn_pos.height)
+                x += Pos.GAP + Pos.WIDTH
+
+        # draw mouse base
+        context.rectangle(830, 660, 244, 90)
+        context.set_source_rgb(0, 0, 0)
+        context.stroke()
+        context.set_font_size(28)
+        context.move_to(900, 720)
+        context.show_text('MOUSE')
+
+        surface.write_to_png(filename)
+
+    def draw_button(self, context, key, x, y, width, height):
+        fn = False
+        if key[:4] == 'XF86':
+            fn = True
+
+        if key in LEGEND:
+            if key == 'modifiers':
+                self.rgb_red(context)
+            elif key == 'group':
+                self.rgb_green(context)
+            elif key == 'layout':
+                self.rgb_cyan(context)
+            elif key == 'window':
+                self.rgb_yellow(context)
+            else:
+                self.rgb_violet(context)
+            context.rectangle(x, y, width, height)
+            context.fill()
+
+        if key in self.modifiers:
+            context.rectangle(x, y, width, height)
+            self.rgb_red(context)
+            context.fill()
+
+        if key in self.keys:
+            k = self.keys[key]
+            context.rectangle(x, y, width, height)
+            self.set_key_color(context, k)
+            context.fill()
+
+            self.show_multiline(context, x + COMMAND_X, y + COMMAND_Y, k)
+
+        context.rectangle(x, y, width, height)
+        context.set_source_rgb(0, 0, 0)
+        context.stroke()
+
+        if fn:
+            key = key[4:]
+            context.set_font_size(10)
+        else:
+            context.set_font_size(14)
+
+        context.move_to(x + BUTTON_NAME_X, y + BUTTON_NAME_Y)
+        context.show_text(self.translate(key))
+
+    def show_multiline(self, context, x, y, key):
```

Comment:
What is this function doing? I think it might help to add a few docstrings when it's not 100% obvious (and most functions are obvious to me, but not this one).

Reply:
cairo doesn't have multiline, so I split lines to fit in button. Wrapping by words